pkg/azureclients/interfaceclient/azure_interfaceclient.go

```diff
@@ -190,20 +190,7 @@ func (c *Client) getVMSSNetworkInterface(ctx context.Context, resourceGroupName
 	)
 
 	result := network.Interface{}
-	computeAPIVersion := ComputeAPIVersion
-	if strings.EqualFold(c.cloudName, AzureStackCloudName) && !c.disableAzureStackCloud {
-		computeAPIVersion = AzureStackComputeAPIVersion
```

Comment:
why is this not necessary? what about AzureStackComputeAPIVersion?

Reply:
I say it is "unnecessary" because I think it won't work? Please check the code below. Actually even this "api-version" in decorator is passed to the request, the `c.apiVersion` (set in `New()`) will the second query pair, which I think is preferred?
https://github.com/kubernetes-sigs/cloud-provider-azure/blob/50d6cd8f087d968af27e5d74fbe01a468ba2848b/pkg/azureclients/armclient/azure_armclient.go#L731-L737

What do you think?

Reply:
then azure stack apiversion should be fixed with this PR

Reply:
I have updated the code. PTAL.

Reply:
if strings.EqualFold(c.cloudName, AzureStackCloudName) && !c.disableAzureStackCloud {
		computeAPIVersion = AzureStackComputeAPIVersion

I mean maybe we need to respect this logic.

Reply:
Yes, I think I did. I moved the logic to `New` because I think the if statement just needs to be checked once.
https://github.com/kubernetes-sigs/cloud-provider-azure/pull/1608/files#diff-fa9aa0f7d9e320680edbcf3fdf7910d9f9fa74c9ed14ae108a74c588f734d012R82-R86